### DISCUSSION TITLE
Trouble with Missing Request Extension for ScyllaDB Session in Axum

### DISCUSSION THREAD

**Discussion by Grifed-source** [tokio-rs/axum#2158] [Q&A] [answered]
Hi everyone,

I'm currently working on a Rust project using Axum for managing my API routes. I have several routes defined in different files and I'm using Axum's 'extensions' to share a ScyllaDB session across these routes.

The idea is to put my ScyllaDB session (ScyllaDB is a NoSQL database) into an extension, so I can access it in every request. However, I'm encountering an issue with this approach.

The error I'm getting is: `'Missing request extension: Extension of type alloc::sync::Arc<scylla::transport::session::Session> was not found'`. Essentially, when Axum receives a request, it's looking for this extension with the ScyllaDB session but it's not finding it.

I suspect that I might have missed adding this extension to the request somewhere in my code. However, I'm not exactly sure where the issue is originating from. Could someone help me understand what might be going wrong and how I can ensure that each request properly includes this extension?

Here's a snippet of my main function where I'm defining my routes and trying to use the ScyllaDB session:

```rust
#[tokio::main]
async fn main() {
    if let Ok(session) = SessionBuilder::new()
        .known_node("162.19.00.00:9042")
        .user("user", "password")
        .build()
        .await
    {
        let app = Router::new()
            .nest("/api", routes())
            .route("/backroute", get(handler));

        let addr = SocketAddr::from(([127, 0, 0, 1], 3001));
        println!("Listening on http://{}", addr);

        if let Err(e) = Server::bind(&addr).serve(app.into_make_service()).await {
            eprintln!("Server error: {}", e);
        }
    } else {
        eprintln!("Unable to create session");
    }
}
```

If you need more code snippets from other files of my project, just let me know. I appreciate any help or insights you could provide. Thank you!

**Comment by davidpdrsn** [comment, marked as answer]
You’re either not adding the extension or extracting an extension with the wrong type. In the code you posted you don’t add the extension anywhere. You need to use https://docs.rs/axum/0.6.20/axum/struct.Extension.html
